lib/bitarray/bit_array.rb

```diff
@@ -0,0 +1,93 @@
+class BitArray
+  attr_reader :field, :reverse_byte, :size
+  include Enumerable
+
+  VERSION = "1.4.0"
+  HEADER_LENGTH = 8 + 1 # QC (@size, @reverse_byte)
+
+  def initialize(size, field = nil, reverse_byte: true)
+    @size = size
+    @field = field || "\0" * (size / 8 + 1)
+    @reverse_byte = reverse_byte
+  end
+
+  # Set a bit (1/0)
+  def []=(position, value)
+    if value == 1
+      @field.setbyte(position >> 3, @field.getbyte(position >> 3) | (1 << (byte_position(position) % 8)))
+    else
+      @field.setbyte(position >> 3, @field.getbyte(position >> 3) & ~(1 << (byte_position(position) % 8)))
+    end
+  end
+
+  # Read a bit (1/0)
+  def [](position)
+    (@field.getbyte(position >> 3) & (1 << (byte_position(position) % 8))) > 0 ? 1 : 0
+  end
+
+  def ==(rhs)
+    @size == rhs.size && @reverse_byte == rhs.reverse_byte && @field == rhs.field
+  end
```

Comment:
Added the ability to compare two `BitArray` objects. This makes it a bit easier to add tests.